### PR TITLE
Documentation for RAPIDS-accelerated Hive UDFs

### LINKED ISSUE #1354: [DOC] Document RAPIDS UDF
**Report needed documentation**
#1351 tracks implementing support for UDFs that provide a RAPIDS execution path. This is a user-visible feature that requires clear documentation on how users can implement their own RAPIDS-accelerated UDFs.

**Describe the documentation you'd like**
There should be a document under the `docs/` directory that is dedicated to describing how UDFs and the RAPIDS Accelerator interact.  That document should either directly describe how to implement a RAPIDS-accelerated UDF or point to a separate, dedicated document for RAPIDS UDFs.  The document should contain pointers to the working examples of RAPIDS UDF implementations.

### PR DESCRIPTION
**Pull request by jlowe**
Adding documentation for how RAPIDS-accelerated UDFs work, tips for how to write one, and pointers to the examples.

Closes #1354

### CODE REVIEW

**Comment by jlowe** [comment]
Marking this as a draft since the documentation links to assets that are provided by #1472 which is still pending.

**Review by tgravescs — DISMISSED** [review]

**Comment by jlowe** [comment]
#1472 has merged, so this is almost ready to go.  It will probably need to me upmerged after @sameerz doc changes for the 0.3 release are done.

File: docs/FAQ.md [hunk 1]
```diff
@@ -168,7 +168,9 @@ can throw at it.
 
 ### How can I run custom expressions/UDFs on the GPU?
 
-We do not currently support this, but we are working on ways to make it possible.
```
Comment (revans2) [inline docs/FAQ.md:171]: Do we want to mention anything about the UDF compiler plugin?

File: docs/rapids-udfs.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+layout: page
+title: RAPIDS-Accelerated User-Defined Functions
+nav_order: 8
+---
+# RAPIDS-Accelerated User-Defined Functions
+
+This document describes how UDFs can provide a RAPIDS-accelerated
+implementation alongside the CPU implementation, enabling the
+RAPIDS Accelerator to perform the user-defined operation on the GPU.
+
```
Comment (revans2) [inline docs/rapids-udfs.md:11]: Again do we want to mention the experimental UDF compiler and point users at that here?

**Review by revans2 — DISMISSED** [review]
Just a few minor possible improvements.

**Review by jlowe — COMMENTED** [review]
@revans2 @tgravescs I updated this to the latest on branch-0.4 which required resolving some merge-conflicts (version.md was deleted) and fixing up the nav order.

@sameerz I also updated RapidsConf.java and TypeChecks.java to include your manual edits from v0.3, otherwise they show up as doc regressions in this PR.

**Review by revans2 — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Review by sameerz — APPROVED** [review]

**Pull request merged**